### PR TITLE
feat(cli): support custom URL slug per locale in docs.yml translations

### PR DESCRIPTION
**Pull request by kafkas**
## Description
Linear ticket: Closes FER-11520

Adds an optional `slug` to `translations` entries in `docs.yml`, letting a localized docs site serve a locale under a custom URL segment — e.g. `lang: ja` with `slug: jp` serves Japanese at `/jp/...` — while the locale keeps its language identity everywhere else (hreflang, `<html lang>`, the language-switcher label, the translated content directory).

This is the **CLI half**. The slug rides the existing publish pipeline: the resolver emits a `localeSlugs` map into the docs config, which is forwarded into the ledger config/manifest, and the docs frontend resolves it at render time (separate fern-platform PR: <link>). No Edge Config or dashboard step.

Locales without a `slug` are completely unaffected — they continue to use their own code as the URL prefix.

## Changes Made
- `docs.yml` schema: add optional `slug` (a URL-safe segment, not the BCP-47 grammar) to translation entries — updated the JSON schema (the validator), the generated serialization + API types, and the zod schema.
- Resolver: emit a `localeSlugs` (locale → slug) map into the docs config so it flows through the ledger publish pipeline into the manifest.
- `fern check` / generation validation: reject a `slug` that would silently mis-route — a recognized locale code (e.g. `ja`), another configured locale's code, or a slug already used by another locale. The reserved-locale set is copied one-to-one from fern-platform's `LOCALE_LABELS`.
- Changelog entry (`feat`).

## Testing
- [x] Unit tests added/updated — `validateTranslationSlugs` (15 cases incl. both conflict edge cases, regional/script codes, case-insensitivity, duplicates), plus translation schema + resolver `localeSlugs` tests.
- [x] Manual testing completed — built the prod CLI and ran `fern check` on a real docs project: `slug: jp` passes; `slug: en` fails with a clear reserved-locale error. Published to a dev FDR and confirmed `/jp/...` serves Japanese end-to-end via the fern-platform PR.

---

<a href="https://app.devin.ai/review/fern-api/fern/pull/16860" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

File: packages/cli/workspace/loader/src/validateTranslationSlugs.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+ * (`packages/fern-docs/components/src/header/language-dropdown-utils.ts`) — that
+ * is the exact set the docs middleware matches URL path prefixes against
+ * (`KNOWN_LOCALES = Object.keys(LOCALE_LABELS)`).
+ *
+ * A translation `slug` override may NOT be one of these codes: the docs
+ * middleware resolves a recognized locale code to its locale at the edge, before
+ * any slug logic runs (it can't read the site config there). So a slug that
+ * collides with a locale code could never take effect and would silently
+ * mis-route. We reject such slugs at `fern check` / generation time instead.
+ */
```
[the remark this answers is not in the document]
Comment (kafkas) [reply packages/cli/workspace/loader/src/validateTranslationSlugs.ts:9]: Out of scope for now, unlikely to be changed soon.

File: packages/cli/workspace/loader/src/loadDocsWorkspace.ts [hunk 2]
```diff
@@ -107,6 +109,21 @@ export async function loadRawDocsConfiguration({
                 code: CliError.Code.ParseError
             });
         }
+
+        // Semantic validation the JSON schema can't express: a translation `slug`
+        // must not collide with a recognized locale code, another configured
+        // locale, or another slug — otherwise it silently mis-routes at the edge.
+        // Fail `fern check` / generation with a clear message instead. (Kept
+        // outside the try above so this message isn't re-wrapped as a parse error.)
+        const slugErrors = validateTranslationSlugs(parsed.translations as RawTranslationEntry[] | undefined);
```
[the remark this answers is not in the document]
Comment (kafkas) [reply packages/cli/workspace/loader/src/loadDocsWorkspace.ts:115]: It does catch malformed slugs currently
